tests/canary/scripts/run_test.sh

```diff
@@ -31,6 +31,8 @@ mkdir -p $E2E_TEST_DIR/.metadata/
 cp metadata-canary $E2E_TEST_DIR/.metadata/
 
 cd $E2E_TEST_DIR
-pytest tests/test_sanity_portforward.py -s -q --metadata .metadata/metadata-canary --region $CLUSTER_REGION --installation_option $INSTALLATION_OPTION
+trap 'python ../canary/scripts/push_stats_to_cloudwatch.py' ERR
```

Comment:
this script already has a trap, see line 15. lets change the trap condition to exit instead of err and move calling the script to that method

similar implementation: https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/test/canary/scripts/run_test.sh

Reply:
changed to EXIT